subjects/devops/orchestrator/README.md

```diff
@@ -0,0 +1,145 @@
+
+### Tips
+
+- Spend time on the theory before rushing into the practice.
+- Read the official documentation.
+- You must understand the k8s components
+
+> Any lack of understanding of the concepts of this project may affect the difficulty of future projects, Take your time to understand all concepts.
+
+> Be curious and never stop searching!
+
+### Architecture
+
+![Architecture](pictures/Architecture.png)
+
+You have to deploy this microservices architecture in a k3s cluster consisting of the following components:
+
+- `inventory-database container` is a PostgreSQL database server that contains your inventory database, it must be accessible via port `5432`.
+- `billing-database container` is a PostgreSQL database server that contains your billing database, it must be accessible via port `5432`.
+- `inventory-app container` is a Node.Js server that contains your inventory-app code running and connected to the inventory database and accessible via port `8080`.
+- `billing-app container` is a Node.Js server that contains your billing-app code running and connected to the billing database and consuming the messages from the RabbitMQ queue, and it can be accessed via port `8080`.
+- `RabbitMQ container` is a RabbitMQ server that contains the queue.
+- `api-gateway-app container` is a Node.Js server that contains your api-gateway-app code running and forwarding the requests to the other services and it's accessible via port `3000`.
+
+> You can use the Dockerfiles you created in the play-with-containers project!
+
+### The cluster
+
+By using k3s in Vagrant you must create 2 virtual machines:
+
+1. `Master`: the master in the k3s cluster.
+
+2. `Agent`: an agent in the k3s cluster.
+
+You must install `kubectl` on your machine to manage your cluster.
+
+The nodes must be connected and available!
+
+```console
+$> kubectl get nodes -A
+NAME                                           STATUS   ROLES    AGE    VERSION
+<master-node>   Ready    <none>   XdXh   vX
+<agent1-node>   Ready    <none>   XdXh   vX
+$>
+```
+
+You must provide a `orchestrator.sh` script that run and create and manage the infrastructure
+
+```console
+$> ./orchestrator create
+cluster created
+$> ./orchestrator start
+cluster started
+$> ./orchestrator stop
+cluster stoped
+$>
+```
+
+### Dockerhub
+
+You will need to push the Docker images for each component to Docker Hub.
+
+> You will use it in your kubernetes manifests.
+
+![Dockerhub-example](pictures/dockerhub-example.jpg)
+
+### Manifests
+
+You should create a YAML Manifest that describes each component or resource of your deployment.
+
+### Secrets
+
+You must store your passwords and credentials as a k8s secrets

```

Comment:
`k8s` is usually written `K8s`, let's be sure we use the right notation in the entire document (including audit)

Reply:
fixed